crates/autopilot/src/domain/settlement/transaction.rs

```diff
@@ -3,31 +3,40 @@
 use {
     super::Settlement,
     crate::{
-        domain::{auction, eth},
+        domain::{auction, competition, eth},
         infra,
     },
 };
 
 /// A transaction that settles a settlement. Interacts with the settlement
 /// contract `settle` function.
+#[derive(Debug)]
 pub struct Tx {
     settlement: Settlement,
     transaction: Transaction,
     #[allow(dead_code)]
     receipt: Receipt,
+    #[allow(dead_code)]
+    auction: super::Auction,
```

Comment:
Seems odd that a struct that's just aggregating a bunch of data is not higher in the directory hierarchy than the individual components. 🤔